runtime/j9vm/java9vmi.c

```diff
@@ -951,93 +952,87 @@ JVM_DefineModule(JNIEnv * env, jobject module, jstring version, jstring location
+
+			PORT_ACCESS_FROM_VMC(currentThread);
+			nameUTF = vmFuncs->copyStringToUTF8WithMemAlloc(
+				currentThread, moduleName, J9_STR_NULL_TERMINATE_RESULT, "", 0, buf, J9VM_PACKAGE_NAME_BUFFER_LENGTH, NULL);
+			if (NULL == nameUTF) {
+				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			} else if ((0 == strcmp(nameUTF, JAVA_BASE_MODULE)) && (classLoader != vm->systemClassLoader)) {
+				vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGLAYERINSTANTIATIONEXCEPTION, "Only bootstrap classloader can define module java.base.");
+			} else {
+				J9Module *j9mod = createModule(currentThread, modObj, classLoader, moduleName);
+				if (NULL != j9mod) {
+					BOOLEAN success = FALSE;
+					UDATA rc = ERRCODE_GENERAL_FAILURE;
 #if J9VM_JAVA9_BUILD >= 156
-				rc = addModuleDefinition(currentThread, j9mod, packages, (U_32) numPackages, version);
+					rc = addModuleDefinition(currentThread, j9mod, packages, (U_32) numPackages, version);
 #else /* J9VM_JAVA9_BUILD >= 156 */
-				rc = addModuleDefinition(currentThread, j9mod, packages, version);
+					rc = addModuleDefinition(currentThread, j9mod, packages, version);
 #endif /* J9VM_JAVA9_BUILD >= 156 */
-
 #if J9VM_JAVA9_BUILD >= 148
-				j9mod->isOpen = isOpen;
+					j9mod->isOpen = isOpen;
 #else /* J9VM_JAVA9_BUILD >= 148 */
-				j9mod->isOpen = FALSE;
+					j9mod->isOpen = FALSE;
 #endif /* J9VM_JAVA9_BUILD >= 148 */
-
-				success = (ERRCODE_SUCCESS == rc);
-
-				if (!success) {
-					vmFuncs->freeJ9Module(vm, j9mod);
-					throwExceptionHelper(currentThread, rc);
-				} else {
-					PORT_ACCESS_FROM_VMC(currentThread);
-					char buf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
-					char *nameUTF = buf;
-
-					nameUTF = vmFuncs->copyStringToUTF8WithMemAlloc(
-						currentThread, moduleName, J9_STR_NULL_TERMINATE_RESULT, "", 0, buf, J9VM_PACKAGE_NAME_BUFFER_LENGTH, NULL);
-					if (NULL == nameUTF) {
-						success = FALSE;
-						goto nativeOOM;
-					}
-
-					/* For "java.base" module setting of jrt URL and patch paths is already done during startup. Avoid doing it here. */
-					if (J9_ARE_ALL_BITS_SET(vm->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED)) {
-						Trc_MODULE_definition(currentThread, nameUTF);
-						if (classLoader == vm->systemClassLoader) {
-							success = vmFuncs->setBootLoaderModulePatchPaths(vm, j9mod, (const char *)nameUTF);
-							if (FALSE == success) {
-								goto nativeOOM;
-							} else {
-								const char* moduleName = "openj9.sharedclasses";
-
-								if (0 == strcmp(nameUTF, moduleName)) {
-									J9VMDllLoadInfo *entry = FIND_DLL_TABLE_ENTRY(J9_SHARED_DLL_NAME);
-
-									if ((NULL == entry)
-										|| (J9_ARE_ALL_BITS_SET(entry->loadFlags, FAILED_TO_LOAD))
-									) {
-										j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_VM_FAILED_TO_LOAD_MODULE_REQUIRED_DLL, J9_SHARED_DLL_NAME, moduleName);
+					success = (ERRCODE_SUCCESS == rc);
+					if (!success) {
+						throwExceptionHelper(currentThread, rc);
```

Comment:
what happened to the `vmFuncs->freeJ9Module(vm, j9mod);` line?

Reply:
Here `success` is `false` hence it executes following:
```
if (FALSE == success) {
	vmFuncs->freeJ9Module(vm, j9mod);
}
```